### PR TITLE
Add baseline_metrics for FileStream to record metrics like elapsed ti…

### LINKED ISSUE #2961: Add baseline_metrics for FileStream to record metrics like elapsed time, record output, etc
**Is your feature request related to a problem or challenge? Please describe what you are trying to do.**

**Describe the solution you'd like**

**Describe alternatives you've considered**

**Additional context**

### PR DESCRIPTION
**Pull request by Ted-Jiang**
…me, record output, etc (#63)

# Which issue does this PR close?

Closes #2961 

 # Rationale for this change

# What changes are included in this PR?

# Are there any user-facing changes?

### CODE REVIEW

**Comment by Ted-Jiang** [comment]
@alamb PTAL 😊

File: datafusion/core/src/physical_plan/file_format/avro.rs [hunk 1]
```diff
@@ -24,17 +24,23 @@ use crate::physical_plan::{
 use arrow::datatypes::SchemaRef;
 
 use crate::execution::context::TaskContext;
+#[cfg(feature = "avro")]
```
Comment (alamb) [inline datafusion/core/src/physical_plan/file_format/avro.rs:27]: Isn't this whole file `#[cfg(feature = "avro")]` ?

In other words, would this work?

```suggestion
```

File: datafusion/core/src/physical_plan/file_format/avro.rs [hunk 2]
```diff
@@ -24,17 +24,23 @@ use crate::physical_plan::{
...
+use crate::physical_plan::metrics::BaselineMetrics;
+use crate::physical_plan::metrics::ExecutionPlanMetricsSet;
 use std::any::Any;
 use std::sync::Arc;
 
 use super::FileScanConfig;
 
 /// Execution plan for scanning Avro data source
 #[derive(Debug, Clone)]
+#[allow(dead_code)]
```
Comment (alamb) [inline datafusion/core/src/physical_plan/file_format/avro.rs:37]: Not sure why this is needed

**Review by alamb — APPROVED** [review]
looks good to me

**Comment by alamb** [comment]
Thanks @Ted-Jiang

**Pull request merged**